Makefile

```
@@ -0,0 +1,93 @@
+.PHONY: $(filter-out vendor bin/tools/psalm/vendor bin/tools/cs-fixer/vendor,$(MAKECMDGOALS))
+
+MYSQL_URL="mysql://root:root@mysql:3306/zenstruck_foundry?charset=utf8"
+POSTGRESQL_URL="postgresql://root:root@postgres:5432/zenstruck_foundry?charset=utf8&serverVersion=13"
+MONGO_URL="mongodb://mongo:mongo@mongo:27017/mongo?compressors=disabled&amp;gssapiServiceName=mongodb&authSource=mongo"
+SQLITE_URL="sqlite://./var/app.db"
+
+ifeq ($(shell docker --help | grep "compose"),)
+	DOCKER_COMPOSE=docker-compose
+else
+	DOCKER_COMPOSE=docker compose
+endif
+
+INTERACTIVE:=$(shell [ -t 0 ] && echo 1)
+ifdef INTERACTIVE
+	DC_EXEC=$(DOCKER_COMPOSE) exec
+else
+	DC_EXEC=$(DOCKER_COMPOSE) exec -T
+endif
+
+DOCKER_PHP=${DC_EXEC} php
+
+.DEFAULT_GOAL := help
+
+UID = $(shell id -u)
+
+DOCKER_IS_UP = $(shell docker-compose ps | grep "Up (healthy)")
+
+ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+$(eval $(RUN_ARGS):;@:)
+
+help:
+	@fgrep -h "###" $(MAKEFILE_LIST) | fgrep -v fgrep | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+validate: fixcs sca test-full ### Run full tests suite
+
+test-full: docker-start vendor ### Run full PHPunit (MySQL + Mongo)
+	@$(eval filter ?= '.')
+	@${DC_EXEC} -e USE_FOUNDRY_BUNDLE=1 -e DATABASE_URL=${MYSQL_URL} -e MONGO_URL=${MONGO_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+
+test-fast: docker-start vendor ### Run PHPunit with SQLite
+	@$(eval filter ?= '.')
+ifeq ($(shell which docker),)
+	@DATABASE_URL=${SQLITE_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+else
+	@${DC_EXEC} -e DATABASE_URL=${SQLITE_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+endif
+
+test-mysql: docker-start vendor ### Run PHPunit with mysql
+	@$(eval filter ?= '.')
+	@${DC_EXEC} -e DATABASE_URL=${MYSQL_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+
+test-postgresql: docker-start vendor ### Run PHPunit with postgreSQL
+	@$(eval filter ?= '.')
+	@${DC_EXEC} -e DATABASE_URL=${POSTGRESQL_URL} php vendor/bin/simple-phpunit --configuration phpunit-dama-doctrine.xml.dist --filter=$(filter)
+
+test-mongo: docker-start vendor ### Run PHPunit with Mongo
+	@$(eval filter ?= '.')
+	@${DC_EXEC} -e MONGO_URL=${MONGO_URL} php vendor/bin/simple-phpunit --filter=$(filter)
```

```suggestion
	@${DC_EXEC} -e MONGO_URL=${MONGO_URL} php vendor/bin/simple-phpunit --configuration phpunit.xml.dist --filter=$(filter)
```

I had an issue with this target because I had my own `phpunit.xml` that phpunit picked up on and used.